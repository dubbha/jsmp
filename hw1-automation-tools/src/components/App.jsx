import React, { PropTypes } from 'react';
import Header from './Header';
import ts from '../typescript.ts';
import raw from '../text.txt';

const App = props => (
  <div>
    <Header>Hello, {props.name}!</Header>
    <div>Hey, {props.name}, App Here!</div>
    <div>{ts.greet()}</div>
    <div>{raw}</div>
  </div>
);

App.propTypes = {
  name: PropTypes.string.isRequired
};

export default App;