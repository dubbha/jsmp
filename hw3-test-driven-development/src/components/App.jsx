import React from 'react';
import PropTypes from 'prop-types';
import calc from '../utils/calc';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      input: props.input ? props.input : '',
      result: calc(props.input)
    };

    this.calculate = this.calculate.bind(this);
    this.digit = this.digit.bind(this);
    this.oper = this.oper.bind(this);
    this.clear = this.clear.bind(this);
    this.submit = this.submit.bind(this);
  }

  render() {
    return (
      <div>
        <div>
          <form className="calc__form" onSubmit={this.submit} autoComplete="off">
            <input
              className="calc__input"
              type="text"
              id="input"
              value={this.state.input}
              onChange={e => this.setState({ input: e.target.value })} />
            <button
              id="equal"
              className="calc__button calc__button-equal"
              onClick={this.calculate}> = </button>
          </form>
          <span id="result" className="calc__result"> {this.state.result} </span>
        </div>
        <div>
          <button className="calc__button" id="d7" onClick={() => this.digit(7)}> 7 </button>
          <button className="calc__button" id="d8" onClick={() => this.digit(8)}> 8 </button>
          <button className="calc__button" id="d9" onClick={() => this.digit(9)}> 9 </button>
          <button className="calc__button" id="div" onClick={() => this.oper('/')}> &#247; </button>
        </div>
        <div>
          <button className="calc__button" id="d4" onClick={() => this.digit(4)}> 4 </button>
          <button className="calc__button" id="d5" onClick={() => this.digit(5)}> 5 </button>
          <button className="calc__button" id="d6" onClick={() => this.digit(6)}> 6 </button>
          <button className="calc__button" id="mult" onClick={() => this.oper('*')}> &#215; </button>
        </div>
        <div>
          <button className="calc__button" id="d1" onClick={() => this.digit(1)}> 1 </button>
          <button className="calc__button" id="d2" onClick={() => this.digit(2)}> 2 </button>
          <button className="calc__button" id="d3" onClick={() => this.digit(3)}> 3 </button>
          <button className="calc__button" id="add" onClick={() => this.oper('+')}> &#43; </button>
        </div>
        <div>
          <button className="calc__button" id="d0" onClick={() => this.digit(0)}> 0 </button>
          <button
            className="calc__button calc__button-clear"
            id="clear"
            onClick={this.clear}> C </button>
          <button className="calc__button" id="pow" onClick={() => this.oper('**')}> ^ </button>
          <button className="calc__button" id="sub" onClick={() => this.oper('-')}> &#8722; </button>
        </div>
      </div>
    );
  }

  calculate() {
    this.setState(() => ({ result: calc(this.state.input) }));
  }

  digit(num) {
    this.setState({ input: this.state.input + num.toString() });
  }

  oper(str) {
    this.setState({ input: this.state.input + str });
  }

  clear() {
    this.setState({
      input: '',
      result: '',
    });
  }

  submit(e) {
    this.calculate();
    e.preventDefault();
  }
}

App.propTypes = {
  input: PropTypes.string
};

export default App;