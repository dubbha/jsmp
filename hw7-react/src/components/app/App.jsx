import React from 'react';
import PropTypes from 'prop-types';
import { Route, NavLink } from 'react-router-dom';
import Dashboard from '../dashboard';
import Heroes from '../heroes';
import HeroDetails from '../hero-details';
import './style.sass';
import '../../assets/react.svg';

class App extends React.Component {
  static propTypes = {
    children: PropTypes.element
  };

  constructor(props) {
    super(props);

    this.state = {
      title: 'React Heroes'
    };
  }

  render() {
    return (
      <div className="app">
        <h1 className="app__header">
          <img src="/react.svg" alt="" className="app__logo" />
          {this.state.title}
        </h1>
        <nav>
          <NavLink to="/dashboard" className="app__navLink" activeClassName="active">Dashboard</NavLink>
          <NavLink to="/heroes" className="app__navLink" activeClassName="active">Heroes</NavLink>
        </nav>

        <Route path="/dashboard" component={Dashboard} />
        <Route path="/heroes" component={Heroes} />
        <Route path="/hero-details/:id" component={HeroDetails} />

      </div>
    );
  }
}

export default App;