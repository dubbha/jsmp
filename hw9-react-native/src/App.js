import React from 'react';
import {
  AppRegistry,
  Image,
  StatusBar,
  StyleSheet,
  TouchableOpacity,
  View,
  CameraRoll,
  Alert,
  Modal,
  ScrollView,
  Button,
  TouchableHighlight,
  Dimensions,
} from 'react-native';
import Camera from 'react-native-camera';

import ModalPreview from './components/ModalPreview';
import ModalGallery from './components/ModalGallery';
import Icon from './components/Icon';

const { width } = Dimensions.get('window');

export default class App extends React.Component {
  constructor(props) {
    super(props);

    this.camera = null;

    this.state = {
      camera: {
        aspect: Camera.constants.Aspect.fill,
        captureTarget: Camera.constants.CaptureTarget.cameraRoll,
        type: Camera.constants.Type.back,
        orientation: Camera.constants.Orientation.auto,
        flashMode: Camera.constants.FlashMode.auto,
      },
      photos: [],
      galleryModalVisible: false,
      previewModalVisible: false,
      previewPhoto: '',
    };
  }

  getPhotos = () => {
    CameraRoll.getPhotos({
      first: 9,
      assetType: 'All'
    })
    .then(r => {
      this.setState({ photos: r.edges });
      this.toggleGalleryModal();
    })
  }

  toggleGalleryModal = () => {
    this.setState({ galleryModalVisible: !this.state.galleryModalVisible });
  }

  togglePreviewModal = () => {
    this.setState({ previewModalVisible: !this.state.previewModalVisible });
  }

  takePicture = () => {
    if (this.camera) {
      this.camera.capture()
        .then((data) => console.log(data))
        .catch(err => console.error(err));
    }
  }

  preview = (uri) => {
    this.setState({ previewPhoto: uri });
    this.togglePreviewModal();
  }

  switchType = () => {
    let newType;
    const { back, front } = Camera.constants.Type;

    if (this.state.camera.type === back) {
      newType = front;
    } else if (this.state.camera.type === front) {
      newType = back;
    }

    this.setState({
      camera: {
        ...this.state.camera,
        type: newType,
      },
    });
  }

  get typeIcon() {
    let icon;
    const { back, front } = Camera.constants.Type;

    if (this.state.camera.type === back) {
      icon = require('./assets/ic_camera_rear_white.png');
    } else if (this.state.camera.type === front) {
      icon = require('./assets/ic_camera_front_white.png');
    }

    return icon;
  }

  switchFlash = () => {
    let newFlashMode;
    const { auto, on, off } = Camera.constants.FlashMode;

    if (this.state.camera.flashMode === auto) {
      newFlashMode = on;
    } else if (this.state.camera.flashMode === on) {
      newFlashMode = off;
    } else if (this.state.camera.flashMode === off) {
      newFlashMode = auto;
    }

    this.setState({
      camera: {
        ...this.state.camera,
        flashMode: newFlashMode,
      },
    });
  }

  get flashIcon() {
    let icon;
    const { auto, on, off } = Camera.constants.FlashMode;

    if (this.state.camera.flashMode === auto) {
      icon = require('./assets/ic_flash_auto_white.png');
    } else if (this.state.camera.flashMode === on) {
      icon = require('./assets/ic_flash_on_white.png');
    } else if (this.state.camera.flashMode === off) {
      icon = require('./assets/ic_flash_off_white.png');
    }

    return icon;
  }

  get collectionsIcon() {
    const icon = require('./assets/ic_collections_white.png');
    return icon;
  }

  render() {
    return (
      <View style={styles.container}>
        <StatusBar
          animated
          hidden
        />
        <Camera
          ref={(cam) => {
            this.camera = cam;
          }}
          style={styles.preview}
          aspect={this.state.camera.aspect}
          captureTarget={this.state.camera.captureTarget}
          type={this.state.camera.type}
          flashMode={this.state.camera.flashMode}
          onFocusChanged={() => {}}
          onZoomChanged={() => {}}
          defaultTouchToFocus
          mirrorImage={false}
        />
        <View style={[styles.overlay, styles.topOverlay]}>
          <Icon
            style={styles.typeButton}
            onPress={this.switchType}
            source={this.typeIcon}>
          </Icon>
          <Icon
            style={styles.flashButton}
            onPress={this.switchFlash}
            source={this.flashIcon}>
          </Icon>
          <Icon
            style={styles.collectionsButton}
            onPress={this.getPhotos}
            source={this.collectionsIcon}>
          </Icon>
        </View>
        <View style={[styles.overlay, styles.bottomOverlay]}>
          <TouchableOpacity
              style={styles.captureButton}
              onPress={this.takePicture}
          >
            <Image
                source={require('./assets/ic_photo_camera_36pt.png')}
            />
          </TouchableOpacity>
          <View style={styles.buttonsSpace} />
        </View>

        <ModalGallery
          visible={this.state.galleryModalVisible}
          onToggleGalleryModal={this.toggleGalleryModal}
          photos={this.state.photos}
          onPreview={this.preview}
          styles={styles}
        >
        </ModalGallery>

        <ModalPreview
          visible={this.state.previewModalVisible}
          onTogglePreviewModal={this.togglePreviewModal}
          previewPhoto={this.state.previewPhoto}
          styles={styles}
        >
        </ModalPreview>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  preview: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  overlay: {
    position: 'absolute',
    padding: 16,
    right: 0,
    left: 0,
    alignItems: 'center',
  },
  topOverlay: {
    top: 0,
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  bottomOverlay: {
    bottom: 0,
    backgroundColor: 'rgba(0,0,0,0.4)',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  captureButton: {
    padding: 15,
    backgroundColor: 'white',
    borderRadius: 40,
  },
  typeButton: {
    padding: 5,
  },
  flashButton: {
    padding: 5,
  },
  collectionsButton: {
    padding: 5,
  },
  buttonsSpace: {
    width: 10,
  },
  modalContainer: {
    paddingTop: 20,
    flex: 1
  },
  scrollView: {
    flexWrap: 'wrap',
    flexDirection: 'row'
  },
});
