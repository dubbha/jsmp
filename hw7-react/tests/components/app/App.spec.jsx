import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from 'components/app';

describe('App', () => {
  it('should render successfully', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
