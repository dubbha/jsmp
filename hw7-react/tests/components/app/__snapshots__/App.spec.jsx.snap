// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render successfully 1`] = `
<div
  className="app"
>
  <h1
    className="app__header"
  >
    <img
      alt=""
      className="app__logo"
      src="/react.svg"
    />
    React Heroes
  </h1>
  <nav>
    <NavLink
      activeClassName="active"
      className="app__navLink"
      to="/dashboard"
    >
      Dashboard
    </NavLink>
    <NavLink
      activeClassName="active"
      className="app__navLink"
      to="/heroes"
    >
      Heroes
    </NavLink>
  </nav>
  <Route
    component={[Function]}
    path="/dashboard"
  />
  <Route
    component={[Function]}
    path="/heroes"
  />
  <Route
    component={[Function]}
    path="/hero-details/:id"
  />
</div>
`;
